tests/trainer/test_checkpoint.py

```diff
@@ -230,275 +182,48 @@ def test_ignore_params(remove_field_paths: List[List[str]], filter_params: List[
+
+    def _assert_weights_equivalent(self, m1: torch.nn.Module, m2: torch.nn.Module):
+        for p1, p2 in zip(m1.parameters(), m2.parameters()):
+            torch.testing.assert_close(p1, p2)
+
+    def get_trainer(self, **kwargs):
+        model = SimpleConvModel()
+        optimizer = torch.optim.Adam(model.parameters())
+
+        return Trainer(
+            model=model,
+            train_dataloader=DataLoader(
+                dataset=RandomImageDataset(),
+                batch_size=8,
+            ),
+            eval_dataloader=DataLoader(
+                dataset=RandomImageDataset(),
+                batch_size=16,
+            ),
+            grad_accum=2,
+            precision='fp32',
+            train_subset_num_batches=5,
+            save_interval='1ep',
+            eval_interval='1ba',
+            save_filename='ep{epoch}.pt',
+            max_duration='2ep',
+            optimizers=optimizer,
+            schedulers=ExponentialScheduler(gamma=0.9),
+            **kwargs,
+        )
 
-    checkpoint_a_folder = 'first'
-    checkpoint_b_folder = 'second'
-    middle_checkpoint = 'ep1.pt'
-    final_checkpoint = 'ep2.pt'
-    latest_checkpoint = composer_trainer_hparams.save_latest_filename.format(rank=dist.get_global_rank())
-    composer_trainer_hparams = get_two_epoch_composer_hparams(composer_trainer_hparams, checkpoint_a_folder)
-    composer_trainer_hparams.run_name = 'big-chungus'
-    second_trainer_hparams = copy.deepcopy(composer_trainer_hparams)
-    # Add object store logger
-    if use_object_store:
+    def get_logger(self, tmp_path: pathlib.Path):
+        """Returns an object store logger that saves locally."""
         remote_dir = str(tmp_path / 'object_store')
         os.makedirs(remote_dir, exist_ok=True)
-        for hparams in [composer_trainer_hparams, second_trainer_hparams]:
-            object_store_logger = ObjectStoreLogger(
-                object_store_cls=LibcloudObjectStore,
-                object_store_kwargs={
-                    'provider': 'local',
-                    'container': '.',
-                    'provider_kwargs': {
-                        'key': remote_dir,
-                    },
-                },
-                num_concurrent_uploads=1,
-                use_procs=use_procs,
-                upload_staging_folder=str(tmp_path / 'staging_folder'),
-            )
-            hparams.loggers = [object_store_logger]
-
-    _test_checkpoint_trainer(composer_trainer_hparams)
-
-    # Create checkpoint in seperate folder to load. Optionally delete original checkpoint by moving it.
-    if delete_local_checkpoint:
-        shutil.move(checkpoint_a_folder, checkpoint_b_folder)
-    else:
-        shutil.copytree(checkpoint_a_folder, checkpoint_b_folder, symlinks=True)
-    # Recreate symlink in new folder
-    new_latest_path = os.path.join(checkpoint_b_folder, latest_checkpoint)
-    os.remove(new_latest_path)
-    os.symlink(final_checkpoint, new_latest_path)
```

Comment:
I also removed this logic here, and removed the use of a `second` folder. Am I missing something here?

Reply:
So what's happening is `autoresume` first checks for local checkpoints, and if there aren't any, checks the object stores for checkpoints to resume from. 

The logic here is to say: 1) backup checkpoint to a second folder so the original trainer can reload it and 2) (optionally) delete the checkpoint so we can test if creating a new trainer reloads from object store if local checkpoint is missing.

We don't need the second folder anymore since we can keep an instantiated copy of the original trainer to use for comparisons. However, we do still need to check if deleting the original file (`delete_local_checkpoint`) works. 

I believe your refactor looks like it handles this correctly to me :)